crates/javy/src/lib.rs

```diff
@@ -117,10 +122,28 @@ pub fn from_js_error(ctx: Ctx<'_>, e: JSError) -> Error {
 pub fn to_js_error(cx: Ctx, e: Error) -> JSError {
     match e.downcast::<JSError>() {
         Ok(e) => e,
-        Err(e) => cx.throw(Value::from_exception(
-            Exception::from_message(cx.clone(), &e.to_string())
-                .expect("creating an exception to succeed"),
-        )),
+        Err(e) => {
+            // In some cases the original error context is lost i.e. we can't
+            // retain the orginal JSError when invoking serde_transcode,
+            // particularly for json::stringify. The process of transcoding will
+            // report the Serializer error, which is totally implementation
+            // dependent, in this case particular to serde_json::Error. To
+            // workaround this, we identify the exception via its string
+            // representation. This is not ideal, but its also fine as it only
+            // happens in the transcoding case.
+            //
+            // Ref: https://github.com/sfackler/serde-transcode/issues/8
+            if e.to_string()
+                .contains("JSError: Exception generated by QuickJS")
```

Comment:
Can we add an automated test case to cover this case?

Reply:
This one is extensively exercised in the stringify 262 test cases. As the comment suggests, it's an edge case of using transcode.